### ISSUE TITLE
support API Gateway logs created via serverless-plugin-stage-variables plugin

### ISSUE THREAD

**Issue by maslakov** [daniel-cottone/serverless-es-logs#197]
At the moment the way to enable API Gateway logs, using **serverless-plugin-stage-variables** is the working one (See this solution https://github.com/serverless/serverless/issues/1918#issuecomment-253362415).
The bug mentioned in the post is fixed and we have a valid Stage object in a resulting CloudFormation template. 

But unfortunately the option **includeApiGWLogs: true** does not work in this case.
From what I see in the logs, this might be because `ServerlessEsLogsPlugin.mergeCustomProviderResources()` is executed before `Serverless: Merged stage variables into ApiGateway Deployment`.

Would be great if this would be possible to make them working together.

**Comment by maslakov** [comment]
This is how **serverless-plugin-stage-variables** is bound to the serverless
https://github.com/svdgraaf/serverless-plugin-stage-variables/blob/0af350187e53d7da3252b0ee3c55823c96254d19/src/index.js#L14

while es-logs uses, `aws:package:finalize:mergeCustomProviderResources`

maybe this would be ok to change it to `after` as well

**Comment by maslakov** [comment]
I doubt it works even now, because the function `addApiGwCloudwatchSubscription` is called only on  **before:aws:deploy:deploy:updateStack**  hook, which according to the logs happens when the template is completely built and uploaded to S3, so the change has no effect.

I tried to move as mentioned before `mergeCustomProviderResources` to "after" hook and this is what I have in logs (have added some logs for observability):
```
Serverless: Merged stage variables into ApiGateway Deployment
Serverless: ServerlessEsLogsPlugin.mergeCustomProviderResources()
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:deploy:deploy
Serverless: [AWS sts 200 0.439s 0 retries] getCallerIdentity({})
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
..
Serverless: ServerlessEsLogsPlugin.beforeAwsDeployUpdateStack()
Serverless: ServerlessEsLogsPlugin.addApiGwCloudwatchSubscription()
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
```

So it's too late

Also had to change the method a bit, so the normal template could be taken
```typescript
    let processorLogicalId: string;
    let template: any;
    if (this.serverless.service.provider.compiledCloudFormationAliasTemplate) {
      template = this.serverless.service.provider.compiledCloudFormationAliasTemplate;
      processorLogicalId = 'EsLogsProcesserAlias';
    } else {
      template = this.serverless.service.provider.compiledCloudFormationTemplate;
      processorLogicalId = 'EsLogsProcesserLambdaFunction';
    }

```

**Comment by daniel-cottone** [comment]
@maslakov This feature already works in conjunction with [serverless-aws-alias](https://github.com/hyperbrain/serverless-aws-alias) plugin. I have not tested this with `serverless-plugin-stage-variables`.

I see that you have raised a PR for this; I would encourage you to test your proposed changes with both plugins. I will try to take a look at your PR this weekend.

**Comment by maslakov** [comment]
I tested it with serverless-plugin-stage-variables - works fine after my fix like described here (https://github.com/serverless/serverless/issues/1918#issuecomment-253362415).

**serverless-aws-alias** unfortunately didn't work for me, no idea why, it could not enable logging in my stack or I had weird JS errors. I will try again, but would be great if you can check (if you have something already prepared), if I have not broken _serverless-aws-alias_ integration. However I wonder how this could work before because of this (https://github.com/daniel-cottone/serverless-es-logs/pull/198/files#diff-f41e9d04a45c83f3b6f6e630f10117feL162) - function name is used instead of ARN.

I personally find the approach with serverless-plugin-stage-variables for API GW logs enabling easier, because it does not change much in the template - only adds Stage object, so I can have manual control over it. In case of aliases it's all much more complicated and for me personally looks like a hack. 

So would be great if both approaches could work.

**Comment by daniel-cottone** [comment]
@maslakov Cloudformation allows you to [reference the function name in place of ARN](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-subscriptionfilter.html) for a cloudwatch subscription filter. I can also confirm that this does work; I am using it 😄 

I will test this with `serverless-aws-alias`, however if it is broken I don't think it is right to break the current working integration in favor of another.

**Comment by maslakov** [comment]
>however if it is broken I don't think it is right to break the current working integration in favor of another.

Of course! I tried to leave them both working. If I didn't manage, I will do my best to fix that.

**Comment by maslakov** [comment]
> I will test this with serverless-aws-alias

@daniel-cottone any news on that?

**Comment by daniel-cottone** [comment]
Sorry, haven't had a chance to look into it. I'll do that this week

**Comment by daniel-cottone** [comment]
@maslakov I am going to close this issue and https://github.com/daniel-cottone/serverless-es-logs/pull/198 since serverless now supports API Gateway logs natively. I've released version `3.0.0` which removes support for any plugin in favor of the native approach.
